### DISCUSSION TITLE
[Question] Are filament table have `when` method or function like Laravel collection?

### DISCUSSION THREAD

**Discussion by zulfikar-ditya** [filamentphp/filament#7764] [Help] [answered]
in some cases we need some condition in our query.
we can use like this in laravel.

```php
      TableName::with('table_name_details')
            ->when(get_current_branch()->is_primary && $request->branch_id, function ($query) use ($request) {
                return $query->where('table_names.branch_id', $request->branch_id);
            })
            ->when(!get_current_branch()->is_primary, function ($query) {
                return $query->where('table_names.branch_id', get_current_branch()->id);
            })
```

Is that any features about this one?
Thanks

**Comment by bezhanSalleh** [comment, marked as answer]
Yes!

**Reply by zulfikar-ditya** [reply]
can you give me short example?

**Reply by danharrin** [reply]
What version are you using

**Reply by zulfikar-ditya** [reply]
3.x

**Reply by danharrin** [reply]
So what is `TableName::with()` supposed to be

**Reply by zulfikar-ditya** [reply]
it  just one to many relation ship, i don't need that this time, i just need the when method

**Reply by danharrin** [reply]
I don't understand, that makes no sense?

**Reply by zulfikar-ditya** [reply]
like this one
https://github.com/protonemedia/laravel-splade/issues/299

**Reply by danharrin** [reply]
I still don't understand. Filament doesn't stop you from using when() on a query, and we have it on most of our own objects as well?
